### ISSUE TITLE
compiling opencv gives error

### ISSUE THREAD

**Issue by jduke32** [bytedeco/javacpp-presets#56]
hi , i when i tried to compile opencv from javacpp , i got error as shown below:
```
C:\Users\Kadir\Desktop\javacpp-presets\opencv\target\classes\org\bytedeco\javacp
p\jniopencv_objdetect.cpp(8865) : warning C4244: 'initializing' : conversion fro
m 'jlong' to 'jint', possible loss of data
C:\Users\Kadir\Desktop\javacpp-presets\opencv\target\classes\org\bytedeco\javacp
p\jniopencv_objdetect.cpp(9036) : warning C4244: 'initializing' : conversion fro
m 'jlong' to 'jint', possible loss of data
Microsoft (R) Incremental Linker Version 12.00.31101.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:jniopencv_objdetect.dll
/ltcg
/dll
/implib:jniopencv_objdetect.lib
/OUT:C:\Users\Kadir\Desktop\javacpp-presets\opencv\target\classes\org\bytedeco\j
avacpp\windows-x86\jniopencv_objdetect.dll
/LIBPATH:C:\Users\Kadir\Desktop\javacpp-presets\opencv/cppbuild/windows-x86/lib/

opencv_world300.lib
jniopencv_objdetect.obj
   Creating library jniopencv_objdetect.lib and object jniopencv_objdetect.exp
jniopencv_objdetect.obj : error LNK2001: unresolved external symbol "public: __t
hiscall cv::DetectionBasedTracker::Parameters::Parameters(void)" (??0Parameters@
DetectionBasedTracker@cv@@QAE@XZ)
C:\Users\Kadir\Desktop\javacpp-presets\opencv\target\classes\org\bytedeco\javacp
p\windows-x86\jniopencv_objdetect.dll : fatal error LNK1120: 1 unresolved extern
als

C:\Users\Kadir\Desktop\javacpp-presets\opencv>
```

**Comment by saudet** [comment]
Right, that seems to be a bug in OpenCV... I guess we'll have to create a
patch for that. Would you be able to look into this? Thanks!

**Comment by jduke32** [comment]
why not.Yes i can test

**Comment by jpsacha** [comment]
@saudet Why do you think it is a bug in OpenCV?

I did a little experiment. The linker complains about 
```
cv::DetectionBasedTracker::Parameters::Parameters(void)
```
I when to the downloaded header `detection_based_tracker.hpp` and commented out constructor in `struct Parameters` that was a potential problem:
```
        struct Parameters
        {
            int maxTrackLifetime;
            int minDetectionPeriod; 
//            Parameters();
        };
```
Then I run `clean` and `install -projects .,opencv`. Now the build completed without errors (not counting some eror character in in comments in `persistance.hpp` that I corrected too). 
The thing that I do not understand is that in both cases (with and without ` Parameters();` commented) the generated wrapper code was almost the same. Not counting different comment and order of declarations the class for `Parameters` was almost identical. Original:
```
        @NoOffset public static class Parameters extends Pointer {
            static { Loader.load(); }
            public Parameters(Pointer p) { super(p); }
            public Parameters(int size) { allocateArray(size); }
            private native void allocateArray(int size);
            @Override public Parameters position(int position) {
                return (Parameters)super.position(position);
            }
        
            public native int maxTrackLifetime(); public native Parameters maxTrackLifetime(int maxTrackLifetime);
            public native int minDetectionPeriod(); public native Parameters minDetectionPeriod(int minDetectionPeriod); //the minimal time between run of the big object detector (on the whole frame) in ms (1000 mean 1 sec), default=0

            public Parameters() { allocate(); }
            private native void allocate();
        }
```
and with `Parameters();` commented (order of declarations rearranged to match above):
```
        public static class Parameters extends Pointer {
            static { Loader.load(); }
            public Parameters(Pointer p) { super(p); }
            public Parameters(int size) { allocateArray(size); }
            private native void allocateArray(int size);
            @Override public Parameters position(int position) {
                return (Parameters)super.position(position);
            }
        
            public native int maxTrackLifetime(); public native Parameters maxTrackLifetime(int maxTrackLifetime);
            public native int minDetectionPeriod(); public native Parameters minDetectionPeriod(int minDetectionPeriod); //the minimal time between run of the big object detector (on the whole frame) in ms (1000 mean 1 sec), default=0

            public Parameters() { allocate(); }
            private native void allocate();
//            Parameters();
        }
```

The only difference is `@NoOffset` annotation (no idea what that actually does).

Can we just fix the issue by commenting out `Parameters();` in `detection_based_tracker.hpp`? When pre-build binaries are used this should have no impact at run-time.

@jduke32 Can you check out if commenting out `Parameters();` works for you?

**Comment by saudet** [comment]
@jpsacha Thanks for starting to look into this! It's a bug in OpenCV because it doesn't give us access to the `cv::DetectionBasedTracker::Parameters::Parameters()` constructor defined in the .cpp file. Maybe we have to put `CV_EXPORT` on the nested class as well? I'm not sure what MSVC expects, but something like that is what needs to be adjusted.

**Comment by jpsacha** [comment]
I create PR that adds building OpenCV binaries from source on Windows, as it is done on other platforms. Before the build two patches are applied to correct issues that were breaking creation of wrappers. One reported earlier, and one more found later. You will need to install Visual Studio 2013 (community edition works) and Windows version of CMake to generate Visual Studio project correctly.

Works fine for me, tested with some sample code from the Cookbook Examples.

**Comment by saudet** [comment]
Fix included in version 1.0. Thanks @jpsacha for the fix and the build!! :+1:
